server/athenian/api/align/goals/measure.py

```diff
@@ -295,27 +309,31 @@ def _parse_team_goal_rows(
             else:
                 jira_filter = JIRAFilter.empty()
 
-            requested_teams.append(
-                RequestedTeamDetails(
-                    team_id=team_id,
-                    goal_id=goal_id,
-                    members=team_member_map[team_id],
-                    repositories=repositories,
-                    jira_filter=jira_filter,
-                ),
+            team_details = RequestedTeamDetails(
+                team_id=team_id,
+                goal_id=goal_id,
+                members=team_member_map[team_id],
+                repositories=repositories,
+                jira_filter=jira_filter,
             )
+            metric_params = _parse_metric_params(
+                team_goal_row[columns[TeamGoal.metric_params.name]],
+            )
+            metric_w_params = MetricWithParams(metric, metric_params)
+            metrics_w_params_by_team[(team_id, goal_id)] = metric_w_params
+            # build a different request for every team
+            # requests will be then simplified by calculate_team_metrics
+            request = TeamMetricsRequest(
```

Comment:
```suggestion
            requests.append(TeamMetricsRequest(
```
(less cognitive load)